building_dialouge_webapp/heat/views.py

```diff
@@ -228,9 +230,136 @@ class Results(SidebarNavigationMixin, TemplateView):
         "next_url": "heat:next_steps",
     }
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        # Daten für die Tabellen
+        scenario1_investments = [20700, 2980, 260, 480, 9620, 36890]
+        scenario1_subsidies = [23158.07]
+        scenario2_investments = [28700, 1980, 260, 410, 9620, 38980, 36890]
+        scenario2_subsidies = [23158.07, 1931.93]
+
+        # Tabellenstruktur erstellen
+        def create_table(title, dynamic_column, values):
+            values = [f"{value:,.2f} €" if isinstance(value, (int, float)) else value for value in values]
+            return pd.DataFrame({
+                title: dynamic_column,
+                "": values
+            })
+
+        # Dynamische Spalten
+        scenario1_dynamic_column_investments = ["Luft-Wärmepumpe", "Wärmespeicher", "Wärmemengenzähler", "Pumpe des Heizsystems", "Lüftungsanlage", "Außenfassade dämmen"]
+        scenario1_dynamic_column_subsidies = ["KfW - Bundesförderung für effiziente Gebäude - Heizungsförderung für Privatpersonen - Wohnungsgebäude (BEG EM) (Nr. 458) (Zuschuss inkl. Klima-Bonus)"]
+
+        scenario2_dynamic_column_investments = ["Luft-Wärmepumpe", "Wärmespeicher", "Wärmemengenzähler", "Pumpe des Heizsystems", "Lüftungsanlage", "Dach dämmen", "Außenfassade dämmen"]
+        scenario2_dynamic_column_subsidies = ["KfW - Bundesförderung für effiziente Gebäude - Heizungsförderung für Privatpersonen - Wohnungsgebäude (BEG EM) (Nr. 458) (Zuschuss inkl. Klima-Bonus)", "KfW - Bundesförderung für effiziente Gebäude - Heizungsförderung für Privatpersonen - Wohnungsgebäude (BEG EM) (Nr. 458) (Effizienz-Bonus)"]
+
+        # Tabellen für Szenario 1
+        df_scenario1_investments = create_table("Investitionskosten", scenario1_dynamic_column_investments, scenario1_investments)
```

Comment:
I would refactor this into a general class `Table` which can be used as parent class for other tables like `InvestmentTable`. The parent class should then provide default attributes and table creation functions, whereby child class would define those attributes in class attributes (like column names, etc.) and values are set during initialization. This makes it more readable.